### PR TITLE
add `matches_type` for array in arg mapper and tests

### PR DESCRIPTION
**Pull request by tcoratger**
## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by Oppen — APPROVED** [review]

File: src/values.rs [hunk 1]
```diff
@@ -170,6 +170,59 @@ impl<T: Into<JitValue>, const N: usize> From<[T; N]> for JitValue {
...
+            (JitValue::Sint32(_), CoreTypeConcrete::Sint32(_)) => Ok(true),
+            (JitValue::Sint64(_), CoreTypeConcrete::Sint64(_)) => Ok(true),
+            (JitValue::Sint128(_), CoreTypeConcrete::Sint128(_)) => Ok(true),
+            (JitValue::EcPoint(_, _), CoreTypeConcrete::EcPoint(_)) => Ok(true),
+            (JitValue::EcState(_, _, _, _), CoreTypeConcrete::EcState(_)) => Ok(true),
+            (JitValue::BoundedInt { .. }, CoreTypeConcrete::BoundedInt(_)) => Ok(true),
+            (JitValue::Null, CoreTypeConcrete::Nullable(_)) => Ok(true),
+            _ => Ok(false),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply src/values.rs]: Yes agreed, just fixed

**Comment by tcoratger** [comment]
@igaray is there anything missing here?

**Review by azteca1998 — DISMISSED** [review]

File: src/arch.rs [hunk 2]
```diff
@@ -194,3 +198,49 @@ impl<'a> AbiArgument for JitValueWithInfoWrapper<'a> {
...
+                },
+                ty: "felt252".into(),
+            }),
+            arena: &bump,
+            registry: &registry,
+        };
+
+        let mut buffer = vec![];
+
+        let _ = jit_value_with_info_wrapper.to_bytes(&mut buffer);
```
[the remark this answers is not in the document]
Comment (tcoratger) [reply src/arch.rs]: Yes just fixed that with unwrap

**Comment by tcoratger** [comment]
@azteca1998 Just fixed the conflicts, let me know if we can merge

**Review by azteca1998 — APPROVED** [review]

**Comment by edg-l** [comment]
The code for jit values changed quite a bit and it has got some tests, closing as this is outdated.

**Pull request closed without merging**